server/src/main/java/org/apache/druid/segment/loading/SegmentLocalCacheManager.java

```diff
@@ -290,6 +291,27 @@ public void storeInfoFile(final DataSegment segment) throws IOException
 
   @Override
   public void removeInfoFile(final DataSegment segment)
+  {
+    final Runnable delete = () -> deleteSegmentInfoFile(segment);
+    final SegmentCacheEntryIdentifier entryId = new SegmentCacheEntryIdentifier(segment.getId());
+    boolean isCached = false;
+    // defer deleting until the unmount operation of the cache entry, if possible, so that if the process stops before
+    // the segment files are deleted, they can be properly managed on startup (since the info entry still exists)
+    for (StorageLocation location : locations) {
+      final SegmentCacheEntry cacheEntry = location.getCacheEntry(entryId);
+      if (cacheEntry != null) {
+        isCached = true;
+        cacheEntry.onUnmount.set(delete);
```

Comment:
Can this race with an actual unmounting? A lock doesn't seem to be acquired at this point in time.